connect-inject/endpoints_controller_test.go

```diff
@@ -2029,123 +2038,347 @@ func TestReconcileUpdateEndpoint(t *testing.T) {
+					ServiceAddress: "1.2.3.4",
+					ServiceMeta: map[string]string{
+						MetaKeyKubeServiceName: "service-updated",
+						MetaKeyKubeNS:          "default",
+						MetaKeyManagedBy:       managedByValue,
+						MetaKeyPodName:         "pod1",
+					},
+				},
+			},
+			expectedProxySvcInstances: []*api.CatalogService{
+				{
+					ServiceID:      "pod1-service-updated-sidecar-proxy",
+					ServiceName:    "service-updated-sidecar-proxy",
+					ServiceAddress: "1.2.3.4",
+					ServiceMeta: map[string]string{
+						MetaKeyKubeServiceName: "service-updated",
+						MetaKeyKubeNS:          "default",
+						MetaKeyManagedBy:       managedByValue,
+						MetaKeyPodName:         "pod1",
+					},
+				},
+			},
+			enableACLs: true,
+		},
 	}
-	// Each test is run with ACLs+TLS (secure) enabled and disabled.
-	for _, secure := range []bool{true, false} {
-		for _, tt := range cases {
-			t.Run(fmt.Sprintf("%s - secure: %v", tt.name, secure), func(t *testing.T) {
-				// The agent pod needs to have the address 127.0.0.1 so when the
-				// code gets the agent pods via the label component=client, and
-				// makes requests against the agent API, it will actually hit the
-				// test server we have on localhost.
-				fakeClientPod := createPod("fake-consul-client", "127.0.0.1", false, true)
-				fakeClientPod.Labels = map[string]string{"component": "client", "app": "consul", "release": "consul"}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			// The agent pod needs to have the address 127.0.0.1 so when the
+			// code gets the agent pods via the label component=client, and
+			// makes requests against the agent API, it will actually hit the
+			// test server we have on localhost.
+			fakeClientPod := createPod("fake-consul-client", "127.0.0.1", false, true)
+			fakeClientPod.Labels = map[string]string{"component": "client", "app": "consul", "release": "consul"}
 
-				// Add the default namespace.
-				ns := corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}}
-				// Create fake k8s client
-				k8sObjects := append(tt.k8sObjects(), fakeClientPod, &ns)
-				fakeClient := fake.NewClientBuilder().WithRuntimeObjects(k8sObjects...).Build()
+			// Add the default namespace.
+			ns := corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}}
+			// Create fake k8s client.
+			k8sObjects := append(tt.k8sObjects(), fakeClientPod, &ns)
+			fakeClient := fake.NewClientBuilder().WithRuntimeObjects(k8sObjects...).Build()
 
-				masterToken := "b78d37c7-0ca7-5f4d-99ee-6d9975ce4586"
-				caFile, certFile, keyFile := common.GenerateServerCerts(t)
-				// Create test consul server, with ACLs+TLS if necessary
-				consul, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
-					if secure {
-						c.ACL.Enabled = true
-						c.ACL.DefaultPolicy = "deny"
-						c.ACL.Tokens.Master = masterToken
-						c.CAFile = caFile
-						c.CertFile = certFile
-						c.KeyFile = keyFile
```

Comment:
Is this removed because it's just a test and what we care about is the ACL behaviour and not whether the traffic is TLS?

Reply:
I'm removing because there isn't any behavior in the endpoints controller that's specific to TLS. Since we're passing the consul client to the endpoints controller struct, so long as that client is configured correctly, everything will work. We're configuring that client in the test as well, and so there's no way this test could ever fail. Since it can't fail, I don't think it's that useful to us.

What triggered it though is that with TLS and ACL table tests, adding new tests was getting a hard. At first, I was thinking of separating those tests and refactoring to make them a bit more readable with the new ACL tests added. But then realized that they can't fail, and so we should probably just remove them.